official/boosted_trees/train_higgs.py

```diff
@@ -0,0 +1,184 @@
+          data[FLAGS.eval_start:FLAGS.eval_start+FLAGS.eval_count])
+
+
+def make_inputs_from_np_arrays(features_np, label_np):
+  """Makes and returns input_fn and feature_columns from numpy arrays.
+
+  The generated input_fn will return the tuple of the dictionary of features
+  and a label, and feature_columns will consist of the list of
+  tf.feature_column.BucketizedColumn.
+
+  Note, for in-memory training, tf.data.Dataset should contain the whole data
+  as a single tensor. Don't use batch.
+
+  Args:
+    features_np: a numpy ndarray (shape=[batch_size, num_features]) for
+        float32 features.
+    label_np: a numpy ndarray (shape=[batch_size, 1]) for labels.
+
+  Returns:
+    input_fn: a function returning a Dataset of feature dict and label.
+    feature_column: a list of tf.feature_column.BucketizedColumn.
+  """
+  num_features = features_np.shape[1]
+  features_np_list = np.split(features_np, num_features, axis=1)
+
+  # Create source feature_columns and bucketized_columns.
+  def get_bucket_boundaries(feature):
+    """Returns bucket boundaries for feature by percentiles."""
+    return np.unique(np.percentile(feature, range(0, 100))).tolist()
+  source_columns = [
+      tf.feature_column.numeric_column(
+          # 1-based feature names.
+          'feature_%02d' % (i + 1), dtype=tf.float32,
+          # Although higgs data have no missing values, in general, default
+          # could be set as 0 or some reasonable value for missing values.
+          default_value=0.0)
+      for i in range(num_features)
+  ]
+  bucketized_columns = [
+      tf.feature_column.bucketized_column(
+          source_columns[i],
+          boundaries=get_bucket_boundaries(features_np_list[i]))
+      for i in range(num_features)
+  ]
+
+  # Make an input_fn that extracts source features.
+  def input_fn():
+    """Returns features as a dictionary of numpy arrays, and a label."""
+    features = {
+        # Give source column name instead for easier track.
+        source_columns[i].name: tf.constant(features_np_list[i])
+        for i in range(num_features)
+    }
+    return tf.data.Dataset.zip((tf.data.Dataset.from_tensors(features),
+                                tf.data.Dataset.from_tensors(label_np),))
+
+  return input_fn, bucketized_columns
+
+
+def main(unused_argv):
+  print('## data loading..')
+  train_data, eval_data = read_higgs_data()
+  print('## data loaded; train: {}{}, eval: {}{}'.format(
+      train_data.dtype, train_data.shape, eval_data.dtype, eval_data.shape))
+  # data consists of one label column and 28 feature columns following.
+  train_input_fn, feature_columns = make_inputs_from_np_arrays(
+      features_np=train_data[:, 1:], label_np=train_data[:, 0:1])
+  eval_input_fn, _ = make_inputs_from_np_arrays(
+      features_np=eval_data[:, 1:], label_np=eval_data[:, 0:1])
+  print('## features prepared. training starts..')
+
+  # TODO(youngheek): consider remove the normalization factor part from l2.
```

Comment:
@ispirmustafa Mustafa, I'm weighing between keeping TODO and removing l2 from the example (so that we could make some changes on the scale later).

WDYT?

Reply:
agreed. let's remove l2. let's not put TODO. you can update the example later.

Reply:
Got it. thanks.